eval.ml

```diff
@@ -286,32 +286,32 @@ let reduce conf =
       pc = pc';
     }
   | Print e ->
-     let v = eval conf e in
-     { conf with
-       trace = v :: conf.trace;
-       pc = pc';
-     }
-  | Osr (e, f, v, l, osr_def) ->
-     let b = get_bool (eval conf e) in
-     if not b then
-       { conf with
-         pc = pc';
-       }
-     else begin
-       let osr_env, heap' = build_osr_frame osr_def conf.env conf.heap in
-       let osr_func = Instr.lookup_fun conf.program f in
-       let osr_version = Instr.get_version osr_func v in
-       let osr_instrs = osr_version.instrs in
-       { conf with
-         pc = resolve osr_instrs l;
-         env = osr_env;
-         heap = heap';
-         instrs = osr_instrs;
-         cur_fun = osr_func.name;
-         cur_vers = osr_version.label;
-         deopt = Some l;
-       }
-     end
+    let v = eval conf e in
+    { conf with
+      trace = v :: conf.trace;
+      pc = pc';
+    }
+  | Osr {cond; target={func;version; label}; map} ->
+    let not_triggered = List.for_all (fun cond -> not (get_bool (eval conf cond))) cond in
```

Comment:
It would be marginally more elegant to define `let triggered` with `List.exists`.